Library/Homebrew/download_strategy.rb

```diff
@@ -450,6 +483,18 @@ def stage
   end
 end
 
+# S3BottleDownloadStrategy downloads bottles from AWS S3.
+# To use it, add ":using => S3BottleDownloadStrategy"
+# to the root_url section of your formula's bottle definition.
+# This download strategy uses AWS access tokens
+# (in the environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY)
+# to sign the request. This strategy is good in a corporate setting,
+# because it lets you use a private S3 bucket as a repo for
+# internal distribution. (It will work for public buckets as well.)
+class S3BottleDownloadStrategy < CurlBottleDownloadStrategy
```

Comment:
It would be good if there was a way the bottle download strategy code could be refactored such that this could use any existing `:using` without needing a code change for each. For example, bottles could be handled by `CurlDownloadStrategy` by default with a `bottle: true` argument or something.

Reply:
Thanks Mike, I'll look into this. Not sure how to do it yet cause I'm anything but a ruby wizard, but I'll see if I can get something to look at :)